brightwind/analyse/analyse.py

```diff
@@ -1156,25 +1161,80 @@ def sector_ratio(wspd_1, wspd_2, wdir, sectors=72, min_wspd=3, direction_bin_arr
+        fig, num = sector_ratio(data['Spd80mN'], data['Spd80mS'], data['Dir78mS'], boom_dir_1=0, boom_dir_2=180,
+                                return_data=True)
+        num
     """
-    wspd_1 = _convert_df_to_series(wspd_1).dropna()
-    wspd_2 = _convert_df_to_series(wspd_2).dropna()
-    wdir = _convert_df_to_series(wdir).dropna()
-
-    sec_rat = _calc_ratio(wspd_1, wspd_2, min_wspd)
-    common_idxs = sec_rat.index.intersection(wdir.index)
-    sec_rat_plot, sec_rat_dist = dist_by_dir_sector(sec_rat.loc[common_idxs], wdir.loc[common_idxs], sectors=sectors,
-                                                    aggregation_method='mean', direction_bin_array=direction_bin_array,
-                                                    direction_bin_labels=None,return_data=True)
-    matplotlib.pyplot.close()
-    sec_rat_dist = sec_rat_dist.rename('Mean_Sector_Ratio').to_frame()
+
+    ws_1 = pd.DataFrame(wspd_1)
+    ws_2 = pd.DataFrame(wspd_2)
+    wd = pd.DataFrame(wdir)
+
+    if len(ws_1.columns) != len(ws_2.columns):
+        raise ValueError('Number of anemometers is uneven. ' +
+                         'Please ensure same number of anemometers in wspd_1 and wspd_2.')
+
+    if len(wd.columns) != 1:
+        if len(wd.columns) != len(ws_1.columns):
+            raise ValueError('Number of anemometers does not match number of wind vanes. ' +
+                             'Please ensure there is one direction vane per anemometer pair or ' +
+                             'include one direcion vane only to be used for all anemometer pairs.')
+
+    keys = range(len(ws_1.columns))
+    sec_rats = {}
+    sec_rats_dists = {}
+    col_names = {}
+
+    for sensor_pair in keys:
+        wspd_1 = _convert_df_to_series(ws_1.iloc[:, sensor_pair]).dropna()
+        wspd_2 = _convert_df_to_series(ws_2.iloc[:, sensor_pair]).dropna()
+
+        if len(wd.columns) == 1:
+            wdir = _convert_df_to_series(wd).dropna()
+        else:
+            wdir = _convert_df_to_series(wd.iloc[:, sensor_pair]).dropna()
+
+        sec_rat = _calc_ratio(wspd_1, wspd_2, min_wspd)
+        sec_rats[sensor_pair] = sec_rat
+        col_names[sensor_pair] = [wspd_1.name, wspd_2.name]
+
+        common_idx = sec_rat.index.intersection(wdir.index)
+
+        sec_rat_plot, sec_rat_dist = dist_by_dir_sector(sec_rat.loc[common_idx], wdir.loc[common_idx], sectors=sectors,
+                                                        aggregation_method='mean',
+                                                        direction_bin_array=direction_bin_array,
+                                                        direction_bin_labels=None, return_data=True)
+        matplotlib.pyplot.close()
+        sec_rat_dist = sec_rat_dist.rename('Mean_Sector_Ratio').to_frame()
+        sec_rats_dists[sensor_pair] = sec_rat_dist
+
+        fig = plt.plot_sector_ratio(sec_ratio=sec_rats, wdir=wdir, sec_ratio_dist=sec_rats_dists, col_names=col_names,
```

Comment:
we should allow to have multiple boom_dir_1 and boom_dir_2. As sensors at different heights can be mounted on booms with different orientation. Do you think that it would be easy to fix this?

Reply:
in analyse.sector_ratio

- Accept int or list of ints as input params
- if list: check len is equal to len wspd_1 (and wspd_2)

- Update doc string
- Add example

in plot.plot_sector_ratio

- if list: check len is equal to something sensible in sec_ratio input
- add a boom counter to iterate through if list: boom_count = 0
- for pair in sec_ratio: if boom_dir is a list, take first element in list boom_dir[0], continue through loop then increment boom_counter += 1 befor next itt. If boom_dir is an int then change nothing

- Update doc string

Reply:
@rach185 @BiancaMorandi this conversation would be better had in the actual issue as that is where things will be better tracked.